src/common/users.ts

```diff
@@ -636,3 +639,65 @@ export const bskySocialUrlMatch =
   /^(?:(?:https:\/\/)?(?:www\.)?bsky\.app\/profile\/)?(?<value>[\w.-]+)(?:\/.*)?$/;
 
 export const portfolioLimit = 500;
+
+const MIN_WORK_EXPERIENCE = 1; // Minimum number of work experiences required for profile completion
+export const isProfileCompleteById = async (
+  con: DataSource,
+  userId: User['id'],
+) => {
+  try {
+    const [user, experiencesCount] = await queryReadReplica(
+      con,
+      ({ queryRunner }) =>
+        Promise.all([
+          queryRunner.manager
+            .getRepository('User')
+            .findOneByOrFail({ id: userId }),
+          queryRunner.manager
+            .getRepository('UserExperience')
+            .count({ where: { userId, type: UserExperienceType.Work } }),
+        ]),
+    );
+
+    const hasCountry =
+      typeof user.flags.country === 'string' && user.flags.country.length >= 2;
+    const hasEnoughExperiences = experiencesCount >= MIN_WORK_EXPERIENCE;
+
+    return hasCountry && hasEnoughExperiences;
```

Comment:
We don't care here about which one is not completed?

Reply:
Not really, for now both are required so we do't need to go further on checks